callbacks.py

```
@@ -0,0 +1,208 @@
+        df_capital, fig_capital_default = figure_manager.get_capital_gain_brkdn(date_range=flow_range)
+
+        return tabs.get_tab_3(df_capital.reset_index(), fig_capital_default)
+
+    @app.callback(
+        Output("capital_fig", "figure"),
+        Input("capital_tbl", "active_cell"),  # fires on click
+        State("year-dropdown", "value"),
+    )
+    def update_capital(active_cell, selected_year):
+        if not active_cell:
+            # no click yet → don’t change graph
```

Ruff (0.16.1)

[warning] Comment contains ambiguous `’` (RIGHT SINGLE QUOTATION MARK). Did you mean ``` (GRAVE ACCENT)?

(RUF003)